mscore/musescore.cpp

```diff
@@ -2990,6 +2993,19 @@ void MuseScore::clipboardChanged()
       }
 
 //---------------------------------------------------------
+//   inputMethodLocaleChanged
+//---------------------------------------------------------
+
+void MuseScore::inputMethodLocaleChanged()
+      {
+      QInputMethod* im = QGuiApplication::inputMethod();
+      _qInputMethodScript = im->locale().script();
+      qDebug("new input method locale: %d", _qInputMethodScript);
+      if (_qInputMethodScript == QLocale::JapaneseScript)
+            qDebug("Might need to handle japanese IME special chars");
+      }
```

Comment:
Hypothetically I don't need to use signal-slot for this, because I could just have that if statement in mscore/keyb.cpp simply query for the language there inside the if statement (mscore->inputMethod()->locale().script() == QLocale::JapaneseScript).  But the nice thing about implementing signal-slot is that now we get a debug message whenever the IME changes script.